File: zettelkasten-front/src/models/File.ts
// src/models/File.ts
import { PartialCard } from "./Card";

export interface File {
  id: number;
  name: string;
  filetype: string;
  path: string;
  filename: string;
  size: number;
  created_by: number;
  updated_by: number;
  card_pk: number;
  is_deleted: boolean;
  created_at: string;
  updated_at: string;
  thumbnail_path: string | null;
  card: PartialCard;
  description?: string;
  extracted_text?: string;
  tags?: string[];
  mimetype?: string;
}

export interface FileTag {
  id: number;
  user_id: number;
  name: string;
  file_count?: number;
}

export interface UploadFileResponse {
  message: string;
  file: File;
}

export interface EditFileMetadataParams {
  name: string;
  card_pk: number;
  description?: string;
}

export interface FileUpdateParams {
  name?: string;
  description?: string;
  card_pk?: number;
}

export interface ImportEpubResponse {
  parent_card_id: number;
  child_card_ids: number[];
  metadata: {
    title: string;
    author: string;
    publisher: string;
    year: string;
    description: string;
  };
}

File: go-backend/models/file.go
package models

import "time"

type File struct {
	ID            int         `json:"id"`
	UserID        int         `json:"user_id"`
	Name          string      `json:"name"`
	Filetype      string      `json:"filetype"`
	Path          string      `json:"path"`
	Filename      string      `json:"filename"`
	Size          int         `json:"size"`
	CreatedBy     int         `json:"created_by"`
	UpdatedBy     int         `json:"updated_by"`
	CardPK        *int        `json:"card_pk,omitempty"`
	IsDeleted     bool        `json:"is_deleted"`
	CreatedAt     time.Time   `json:"created_at"`
	UpdatedAt     time.Time   `json:"updated_at"`
	ThumbnailPath *string     `json:"thumbnail_path"`
	Card          PartialCard `json:"card"`
	Description   *string     `json:"description,omitempty"`
	ExtractedText *string     `json:"extracted_text,omitempty"`
	Snippet       *string     `json:"snippet,omitempty"`       // Populated on search: text around the match
	SnippetField  *string     `json:"snippet_field,omitempty"` // name | description | content | tag
	Tags          []string    `json:"tags,omitempty"`          // Populated on read
}

type EditFileMetadataParams struct {
	Name        string  `json:"name"`
	CardPK      *int    `json:"card_pk,omitempty"`
	Description *string `json:"description,omitempty"`
}

type FileUpdateParams struct {
	Name        *string `json:"name,omitempty"`
	Description *string `json:"description,omitempty"`
	CardPK      *int    `json:"card_pk,omitempty"`
}

type UploadFileResponse struct {
	Message string `json:"message"`
	File    File   `json:"file"`
}

type UploadFileParams struct {
	Filename    string `json:"filename"`
	ContentType string `json:"content_type"`
}
